File: Python solutions/226 invert binary tree.py
# https://leetcode.com/problems/invert-binary-tree
# TC: O(n) SC: O(1)
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root != None:
            root.left, root.right = root.right, root.left
            self.invertTree(root.left)
            self.invertTree(root.right)
        return root

File: Cpp solutions/226 invert binary tree.cpp
// https://leetcode.com/problems/invert-binary-tree
// TC:O(n) SC:O(1)
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root != NULL){
            TreeNode *temp = root->left;
            root->left = root->right;
            root->right = temp;
            invertTree(root->left);
            invertTree(root->right);
        }
        return root;
    }
};